src/Shared/FileUtilities.cs

```diff
@@ -1031,27 +1031,72 @@ internal static string MakeRelative(string basePath, string path)
             ErrorUtilities.VerifyThrowArgumentNull(basePath, nameof(basePath));
             ErrorUtilities.VerifyThrowArgumentLength(path, nameof(path));
 
-            if (basePath.Length == 0)
-            {
-                return path;
-            }
+            string fullBase = Path.GetFullPath(basePath);
+            string fullPath = Path.GetFullPath(path);
 
-            Uri baseUri = new Uri(EnsureTrailingSlash(basePath), UriKind.Absolute); // May throw UriFormatException
+            string[] splitBase = fullBase.Split(new char[] { Path.DirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries);
+            string[] splitPath = fullPath.Split(new char[] { Path.DirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries);
 
-            Uri pathUri = CreateUriFromPath(path);
+            ErrorUtilities.VerifyThrow(splitPath.Length > 0, "Cannot call MakeRelative on a path of only slashes.");
 
-            if (!pathUri.IsAbsoluteUri)
+            // On a mac, the path could start with any number of slashes and still be valid. We have to check them all.
+            int indexOfFirstNonSlashChar = 0;
+            while (path[indexOfFirstNonSlashChar] == Path.DirectorySeparatorChar)
             {
-                // the path is already a relative url, we will just normalize it...
-                pathUri = new Uri(baseUri, pathUri);
+                indexOfFirstNonSlashChar++;
+            }
+            if (path.IndexOf(splitPath[0]) != indexOfFirstNonSlashChar)
+            {
+                // path was already relative so just return it
+                return FixFilePath(path);
             }
 
-            Uri relativeUri = baseUri.MakeRelativeUri(pathUri);
-            string relativePath = Uri.UnescapeDataString(relativeUri.IsAbsoluteUri ? relativeUri.LocalPath : relativeUri.ToString());
+            int baseI = 0;
+            int pathI = 0;
+            while (true)
+            {
+                if (baseI == splitBase.Length)
+                {
+                    if (pathI == splitPath.Length)
+                    {
+                        return ".";
+                    }
+                    break;
+                }
+                else if (pathI == splitPath.Length)
+                {
+                    break;
+                }
+                else if (splitBase[baseI].Equals(splitPath[pathI], PathComparison))
+                {
+                    baseI++;
+                    pathI++;
+                }
+                else
+                {
+                    break;
+                }
+            }
```

Comment:
I would find the following much easier to read. As long as it's equivalent, please double-check:

```suggestion
            while (baseI < splitBase.Length && pathI < splitPath.Length && splitBase[baseI].Equals(splitPath[pathI], PathComparison))
            {
                    baseI++;
                    pathI++;
            }
            if (baseI == splitBase.Length && pathI == splitPath.Length)
            {
                   return ".";
            }
  }
```